### ISSUE TITLE
Neural Net Debugging Capabilities 

### ISSUE THREAD

**Issue by sebderhy** [airctic/icevision#607]
After I train an object detector, I would love to have tools to better understand how it takes its decisions, why does it fail etc... 
Here are some papers/resources that could be relevant to integrate/implement:

- Many of interesting implementations can be found for example [on this repo](https://github.com/utkuozbulak/pytorch-cnn-visualizations/) 
- [Grad-CAM](https://github.com/fastai/fastbook/blob/master/18_CAM.ipynb)
- [Visualizing the Loss Landscape of Neural Nets](https://github.com/tomgoldstein/loss-landscape)
- [Stanford Lecture on Understanding CNNs](http://cs231n.stanford.edu/slides/2020/lecture_13.pdf)
- [Captum](https://captum.ai/)

... and I'm sure there are many other resources!

**Comment by FraPochetti** [comment]
Not enough bandwidth to implement
